core/mondoo-okta-security.mql.yaml

```diff
@@ -1,379 +1,1985 @@
+    impact: 100
+    docs:
+      desc: |
+        Configure network blocklisting to deny access from known malicious IP addresses or locations from your Okta org.
+
+        Admins can block access to their Okta org to IP addresses coming from network zones, IP zones, and dynamic zones. Network zones contain a list of IP addresses, and dynamic zones contain a list of locations, ASNs, or IP types. Okta doesn't allow blocklisted IP addresses to access any of your org's URLs. Okta blocks these requests before any type of policy evaluation occurs.
+
+        ### Okta recommends
+
+        Block any known untrusted IP addresses, locations, or proxy servers to limit access to your org. If your org uses IP Trust for network zones, Okta also recommends blocking any IP addresses that are identified as a Tor anonymizer proxy. Only add IP addresses or locations that aren't associated with legitimate user activity.
+
+        ### Security impact
+
+        Moderate
+
+        ## User impact
+
+        Low
+
+        Legitimate users within your org see no change in behavior. Clients connecting from blocked network zones see a 403 (access denied) error.
+    variants:
+      - uid: mondoo-okta-security-threatinsight-block-suspicous-ip-addresses-terraform-hcl
+      - uid: mondoo-okta-security-threatinsight-block-suspicous-ip-addresses-terraform-plan
+      - uid: mondoo-okta-security-threatinsight-block-suspicous-ip-addresses-terraform-state
+      - uid: mondoo-okta-security-threatinsight-block-suspicous-ip-addresses-runtime
+  - uid: mondoo-okta-security-threatinsight-block-suspicous-ip-addresses-runtime
+    title: Enable Okta ThreatInsight to block suspicious IP addresses
+    impact: 100
+    filters: asset.platform == "okta-org"
+    mql: |
+      okta.networks.any( usage == "BLOCKLIST" && gateways.length > 0 )
+    docs:
+      remediation: 
+        - id: console 
+          desc: |
+            ### Block specific IP addresses
+            Block specific IP addresses to deny access to your Okta org.
+
+            1. In the Admin Console, go to SecurityNetworks.
+            2. In the list of zones, click Edit for the BlockedIpZone network zone.
+            3. Select Block access from IPs matching conditions listed in this zone.
+            4. Click Save.
+            
+            ### Block IP addresses in a dynamic zone
+            Block IP addresses in a dynamic zone from accessing your Okta org.
+
+            1. In the Admin Console, go to SecurityNetworks.
+            2. Click Add Zone > Dynamic Zone.
+            3. Define a location or proxy type.
+            4. Select Block access from IPs matching conditions listed in this zone.
+            5. Click Save.
+
+            ### Block Tor anonymizer proxy IP addresses
+            Block IP addresses identified as a Tor anonymizer proxy from accessing your Okta org.
+
+            1. In the Admin Console, go to SecurityNetworks.
+            2. Click Add Zone > Dynamic Zone.
```

Comment:
```suggestion
            2. Select Add Zone > Dynamic Zone.
```

Reply:
@tas50 these are not our steps. they are from Okta's docs. They are copy/pasted.